docs/architecture/adr-010-posts-module.md

```diff
@@ -0,0 +1,497 @@
+  optional Entities entities = 5;
+
+  // Author of the post
+  required string author = 6;
+}
+
+// MsgCreatePostResponse defines the Msg/EditPost response type.
+message MsgEditPostResponse {
+  // Edit date of the post
+  required google.protobuf.Timestamp edit_date = 1;
+}
+
+// MsgAddPostAttachment represents the message that should be
+// used when adding an attachment to post
+message MsgAddPostAttachment {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+  
+  // Id of the post to which to add the attachment
+  required uint64 post_id = 2;
+  
+  // Attachment to be added to the post 
+  required Attachment attachment = 3;
+
+  // Author of the post
+  required string author = 4;
+  
+  message Attachment {
+    oneof sum {
+      Poll poll = 1;
+      Media media = 2;
+    }
+  }
+}
+
+// MsgAddPostAttachmentResponse defines the Msg/AddPostAttachment response type.
+message MsgAddPostAttachmentResponse {
+  // New id of the uploaded attachment 
+  required uint32 attachment_id = 1;
+
+  // Edit date of the post
+  required google.protobuf.Timestamp edit_date = 2;
+}
+
+// MsgRemovePostAttachment represents the message to be used when 
+// removing an attachment from a post
+message MsgRemovePostAttachment {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+
+  // Id of the post from which to remove the attachment
+  required uint64 post_id = 2;
+  
+  // Id of the attachment to be removed
+  required uint32 attachment_id = 3;
+
+  // User that is removing the attachment
+  required string signer = 4;
+}
+
+// MsgRemovePostAttachmentResponse defines the 
+// Msg/RemovePostAttachment response type.
+message MsgRemovePostAttachmentResponse {
+  // Edit date of the post
+  required google.protobuf.Timestamp edit_date = 1;
+}
+
+// MsgDeletePost represents the message used when deleting a post.
+message MsgDeletePost {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+
+  // Id of the post to be deleted
+  required uint64 post_id = 2;
+  
+  // User that is deleting the post
+  required string signer = 3;
+}
+
+// MsgDeletePostResponse represents the Msg/DeletePost response type
+message MsgDeletePostResponse {}
+
+// MsgAnswerPoll represents the message used to answer a poll
+message MsgAnswerPoll {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+
+  // Id of the post that contains the poll to be answered
+  required uint64 post_id = 2;
+  
+  // Id of the poll to be answered 
+  required uint32 poll_id = 3;
+  
+  // Indexes of the answer inside the ProvidedAnswers array
+  repeated uint32 answers_indexes = 4;
+  
+  // Address of the user answering the poll
+  required string signer = 5;
+}
+
+// MsgAnswerPollResponse represents the MSg/AnswerPoll response type
+message MsgAnswerPollResponse {}
+```
+
+### `Query` Service
+```protobuf
+// Query defines the gRPC querier service
+service Query {
```

Comment:
It would be useful to have a `PollResult` to query the result of a poll.
[Twitter Poll](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/poll)

Reply:
Added :+1: